File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/command/ConcrectCommand.java
package jv.com.tencent.cain.design_pattern.command;

/**
 * @author cainjiang
 * @date 2020/7/20
 */
public class ConcrectCommand extends AbstractCommand {

    public ConcrectCommand(Receiver receiver) {
        super(receiver);
    }

    @Override
    void execute() {
        /**
         * 具体的执行
         */
        receiver.action();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/command/ConcrectCommand.kt
package kt.com.tencent.cain.desigin_pattern.command

/**
 * @author cainjiang
 * @date 2020/7/20
 */
class ConcrectCommand(receiver: Receiver) : BaseCommand(receiver) {
    override fun execute() {
        receiver.action()
    }
}